bids/variables/io.py

```diff
@@ -147,7 +147,8 @@ def _load_time_variables(layout, dataset=None, columns=None, scan_length=None,
         try:
             import nibabel as nb
             img = nb.load(img_f)
-            duration = img.shape[3] * img.header.get_zooms()[-1]
+            tr = layout.get_metadata(img_f)['RepetitionTime']
```

Comment:
what if it is not there?
The bible [describes](https://bids-specification.readthedocs.io/en/stable/04-modality-specific-files/01-magnetic-resonance-imaging-data.html#required-fields) it as REQUIRED but then it says that it is "mutually exclusive" with `VolumeTiming`.  I do not think that if a varying VolumeTiming is supported anywhere here but then it either must blow  with a meaningful message if RepetitionTime is absent or look into VolumeTiming and verify that it is equispaced/deduce the TR, or fallback to lookup in .nii.gz
This echoes with another comment +1'ing Satra that .json ideally takes precendence since apparently with VolumeTiming it might be quite obscure case and pybids better blows if finds notably variable timing instead of plowing through

PS FWIW, I saw no dataset with `VolumeTiming` in metadata extracted from openneuro datasets, so it is more of a "future proofing" I guess